src/special/mocha.js

```diff
@@ -1,41 +1,82 @@
 import fs from 'fs';
 import path from 'path';
-import lodash from 'lodash';
 import requirePackageName from 'require-package-name';
 import { getScripts } from '../utils';
 
-function getOpts(script) {
-  const argvs = script.split(' ').filter(argv => argv);
+const knownReporters = [
+  'dot', 'doc', 'tap', 'json', 'html', 'list',
+  'min', 'spec', 'nyan', 'xunit', 'markdown', 'progress',
+  'landing', 'json-stream',
+];
+
+function getOptsConfig(root, config) {
+  const argvs = config.split(/\s+/);
   const optsIndex = argvs.indexOf('--opts');
-  return optsIndex !== -1 ? argvs[optsIndex + 1] : null;
+
+  if (optsIndex === -1) {
+    return null;
+  }
+
+  const optsPath = argvs[optsIndex + 1];
+
+  if (!optsPath) {
+    return null;
+  }
+
+  return fs.readFileSync(path.resolve(root, '..', optsPath), 'utf-8');
 }
 
-function getRequires(content, deps) {
-  return content
-    .split('\n')
-    .map(line => line.trim())
-    .filter(line => line.indexOf('--require ') === 0)
-    .map(line => line.substring('--require '.length).trim())
+function getDependencies(content, deps) {
+  const lines = content.split(/\s+/);
+  const result = [];
+
+  lines.forEach((line, idx) => {
+    if (line === '--require') {
+      const val = lines[idx + 1];
+      if (val && !val.startsWith('--')) {
+        result.push(val);
+      }
+    }
+    if (line === '--reporter') {
+      const val = lines[idx + 1];
+      if (val && !val.startsWith('--') && !knownReporters.includes(val)) {
+        result.push(val);
+      }
+    }
+  });
+
+  return result
     .map(requirePackageName)
-    .filter(name => deps.indexOf(name) !== -1);
+    .filter((v, k, arr) => arr.indexOf(v) === k)
+    .filter(name => deps.includes(name));
```

Comment:
this seems weird, maybe it shouldn't be here? it was there before but i wasn't sure why...

Reply:
it reads as if it will filter out any dependencies which aren't in our package.json? don't we want to know about those? i left it in because it was there before but seems pretty questionable.